### PR TITLE
Feature/wid 273

### PR DESCRIPTION
**Pull request by brampauwelyn**
### Added
- added yaml file with fallback images
- show fallback image if setting is enabled

---
Ticket: [https://jira.uitdatabank.be/browse/WID-273](https://jira.uitdatabank.be/browse/WID-273)

### CODE REVIEW

File: src/Widget/Twig/TwigPreprocessor.php [hunk 1]
```diff
@@ -368,6 +368,36 @@ public function preprocessEventDetail(Event $event, string $langcode, array $set
...
+        if (!$settings['enabled']) {
+            return '';
+        }
+
+        if ($settings['type'] === 'uit') {
+            return $this->request->getScheme() . '://media.uitdatabank.be/static/uit-placeholder.png';
+        }
+
+        if ($settings['type'] === 'theme') {
+            $fallbackImages = Yaml::parse(file_get_contents(__DIR__ . '/../../../fallback_images.yml'));
```
Comment (brampauwelyn) [inline src/Widget/Twig/TwigPreprocessor.php:389]: @konrad126 / @JonasVHG 
Not sure about this line.
Would it have an impact on the performance to read this file for every event?

Comment (jdrieghe) [reply src/Widget/Twig/TwigPreprocessor.php:389]: You could always store it in a local property and parse the yml if the local property is empty

File: src/Widget/Twig/TwigPreprocessor.php [hunk 2]
```diff
@@ -70,6 +70,11 @@ class TwigPreprocessor
      */
     protected $curatorenClient;
 
+    /**
+     * @var array
+     */
+    public $fallbackImages;
```
Comment (jdrieghe) [inline src/Widget/Twig/TwigPreprocessor.php:76]: Please make this `private`. There's no need for external classes to have access to this

File: src/Widget/Twig/TwigPreprocessor.php [hunk 3]
```diff
@@ -386,9 +394,10 @@ public function getDefaultImage($settings, $eventTypeId)
         }
 
         if ($settings['type'] === 'theme') {
-            $fallbackImages = Yaml::parse(file_get_contents(__DIR__ . '/../../../fallback_images.yml'));
 
-            foreach ($fallbackImages as $fallbackImage) {
+            $this->fallbackImages = (empty($this->fallbackImages)) ? Yaml::parse(file_get_contents(__DIR__ . '/../../../fallback_images.yml')) : $this->fallbackImages;
```
Comment (jdrieghe) [inline src/Widget/Twig/TwigPreprocessor.php:397]:
```suggestion
            if (empty($this->fallbackImages)) {
                $this-fallbackImages = Yaml::parse(file_get_contents(__DIR__ . '/../../../fallback_images.yml'));
            }
```

**Pull request merged**